lib/cocoapods/installer/analyzer.rb

```diff
@@ -542,305 +617,36 @@ def sources
-            deployment_target = Version.new(target.deployment_target)
+      def check_platform_specifications
+        unless config.integrate_targets?
+          podfile.target_definition_list.map do |target_definition|
+            unless target_definition.platform
+              raise Informative, 'It is necessary to specify the platform in the Podfile if not integrating.'
+            end
           end
         end
-
-        target_definition.set_platform(name, deployment_target)
-        Platform.new(name, deployment_target)
-      end
-
-      # @return [Platform] The platform for the library.
-      #
-      # @note   This resolves to the lowest deployment target across the user
-      #         targets.
-      #
-      # @todo   Is assigning the platform to the target definition the best way
-      #         to go?
-      #
-      def compute_archs_for_target_definition(target_definition, user_targets)
-        archs = []
-        user_targets.each do |target|
-          target_archs = target.common_resolved_build_setting('ARCHS')
-          archs.concat(Array(target_archs))
-        end
-
-        archs = archs.compact.uniq.sort
-        UI.message('Using `ARCHS` setting to build architectures of ' \
-                   "target `#{target_definition.label}`: " \
-                   "(`#{archs.join('`, `')}`)")
-        archs.length > 1 ? archs : archs.first
       end
 
-      # Precompute the platforms for each target_definition in the Podfile
+      # Precompute information for each target_definition in the Podfile
       #
       # @note The platforms are computed and added to each target_definition
       #       because it might be necessary to infer the platform from the
       #       user targets.
       #
-      # @return [void]
+      # @return [Hash{TargetDefinition => TargetInspectionResult}]
       #
-      def compute_target_platforms
+      def inspect_targets_to_integrate
+        inspection_result = {}
         UI.section 'Inspecting targets to integrate' do
-          podfile.target_definition_list.each do |target_definition|
-            if config.integrate_targets?
-              project_path = compute_user_project_path(target_definition)
-              user_project = Xcodeproj::Project.open(project_path)
-              targets = compute_user_project_targets(target_definition, user_project)
-              compute_platform_for_target_definition(target_definition, targets)
-              archs = compute_archs_for_target_definition(target_definition, targets)
-              @archs_by_target_def[target_definition] = archs
-            else
-              unless target_definition.platform
-                raise Informative, 'It is necessary to specify the platform in the Podfile if not integrating.'
-              end
-            end
+          podfile.target_definition_list.map do |target_definition|
+            inspection_result[target_definition] = TargetInspector.new(target_definition).inspect!
           end
```

Comment:
Maybe it's a good idea to rename that method as it is very near to stdlib's debugging helper `#inspect`.